pac4j-http/src/main/java/org/pac4j/http/client/direct/DirectDigestAuthClient.java

```diff
@@ -0,0 +1,73 @@
+package org.pac4j.http.client.direct;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.RequiresHttpAction;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.http.credentials.CredentialUtil;
+import org.pac4j.http.credentials.DigestCredentials;
+import org.pac4j.http.credentials.authenticator.DigestAuthenticator;
+import org.pac4j.http.credentials.extractor.DigestAuthExtractor;
+import org.pac4j.http.profile.creator.ProfileCreator;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Random;
```

Comment:
This import used anywhere?

I imagine you have many other unnecessary imports. Make sure you clean up everything that is not used.